tempesta_fw/http.c

```diff
@@ -1162,23 +1215,38 @@ tfw_http_conn_collect(TfwSrvConn *srv_conn, struct list_head *sch_queue,
  * with minimal effort.
  */
 static void
-tfw_http_conn_resched(struct list_head *sch_queue, struct list_head *equeue)
+tfw_http_conn_resched(struct list_head *sch_queue, struct list_head *equeue,
+		      TfwSrvConn *prev_conn)
 {
 	TfwSrvConn *sch_conn;
 	TfwHttpReq *req, *tmp;
+	TfwServer *srv = (TfwServer *)prev_conn->peer;
 
 	/*
 	 * Process the complete queue and re-schedule all requests
-	 * to other servers/connections.
+	 * to other servers/connections. Health monitoring requests
+	 * must be re-scheduled to the same server (other servers
+	 * may not have enabled health monitor).
```

Comment:
Requests in a connection are rescheduled (which is different from resending) when all attempts to reconnect have failed (there's a configurable number of attempts). Attempts to reconnect may take time which may overlap with health monitoring. That means that by the time a health monitoring request is rescheduled, either the relevant requests (to a configured URL) or another health monitoring request may have went to the server through another connection. Perhaps it's best to check for that before rescheduling a health monitoring request.

On the other hand, rescheduling and sending these "outdated" health monitoring requests won't hurt. If that's the idea, then perhaps it's best to document it.